### DISCUSSION TITLE
Erroneously synced too many directories...

### DISCUSSION THREAD

**Discussion by JosN1** [abraunegg/onedrive#1695] [Q&A] [answered]
My installation should be synchronizing only one subfolder to my laptop. It was not syncing (Fedora 35), so I **thought** I was manually forcing a synchronization. Instead, I had begun to sync all of my OneDrive files. I interrupted the downloading before it completed. I would now like to reconfigure the app so that only that one subfolder is synchronized and no other OneDrive folders are within the sync scope. How can I do this? (I would like to avoid uninstalling, reinstalling, and reconfiguring.) Help, please.

**Comment by abraunegg** [comment, marked as answer]
@JosN1 
To resolve your issue:
* Use `onedrive --display-config` to understand your configuration
* Use `onedrive --synchronize --dry-run` to validate your configuration is doing what you want it to do
* Review https://github.com/abraunegg/onedrive/blob/master/docs/USAGE.md to configure your client

**Comment by JosN1** [comment]
Finally getting to this.... Thank you for your earlier response.

If I want to sync only OneDrive/Documents/'specific-directory', is there a way to have the onedrive client look only to that directory for syncing, or must I exclude all other directories in my client config?

I performed a selective directory sync and it worked as expected, but when I perform a dry run it syncs the whole OD footprint.

________________________________

**Comment by abraunegg** [comment]
@JosN1 
> If I want to sync only OneDrive/Documents/'specific-directory', is there a way to have the onedrive client look only to that directory for syncing, or must I exclude all other directories in my client config?

Please read  https://github.com/abraunegg/onedrive/blob/master/docs/USAGE.md in regards to the following options:
* 'sync_list'
* `--single-directory`


> I performed a selective directory sync and it worked as expected, but when I perform a dry run it syncs the whole OD footprint

Potentially you were not using the right commands, for example:
**Dry run + Single Directory Sync:** Use `onedrive --synchronize --single-directory 'path/to/directory' --dry-run`
**Single Directory Sync:** Use `onedrive --synchronize --single-directory 'path/to/directory' `

The only difference between the two is the addition | removal of `--dry-run`
